File: src/types/dtos/exerciseDTO.ts
export type ExerciseType = "text" | "multiple-choice" | "image";

export interface ExerciseBaseDTO {
  id: string;
  question: ExerciseComponent[];
  answer: ExerciseComponent[]; // only one answer is allowed, but Image components could be necessary
  lastEdited: Date;
  createdOn: Date;
  documentationUrl?: string;
  authorId: string;
}

export type ExerciseComponent =
  | ExerciseTextComponent
  | ExerciseImageComponent
  | ExerciseMultipleChoiceComponent;

export interface ExerciseTextComponent {
  type: "text";
  content: string;
}

export interface ExerciseImageComponent {
  type: "image";
  imageUrl: string;
}

export interface ExerciseMultipleChoiceComponent {
  type: "multiple-choice";
  choices: string[];
  correctChoice: string;
}

export type ExerciseTyp = "image" | "text" | "multiple-choice";

File: api/dtos/ExerciseDTO.py
from typing import Optional
from pydantic import BaseModel

class ExerciseBaseDTO(BaseModel):
    id:Optional[str] = None
    type:str

class ExerciseNormalDTO(ExerciseBaseDTO):
    question:str
    answer:str
    

class ExerciseMultipleChoiceDTO(ExerciseBaseDTO):
    question:str
    choices:list[str]
    answer:str
